### PR TITLE
[PIE-2333] Fix trace replay block transactions

### PR DESCRIPTION
**Pull request by matkt**
## PR description

https://pegasys1.atlassian.net/browse/PIE-2333

Some fields (creationMethod, mem) are missing on the API trace_replayBlockTransactions 

- Add creationMethod field
- Resolve missing mem field for CALL and DELEGATECALL with RETURN
- Resolve invalid cost value for STATICCALL 
- Add new tests

### CODE REVIEW

File: ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/results/tracing/flat/FlatTraceGenerator.java [hunk 1]
```diff
@@ -428,21 +431,26 @@ private static void addAdditionalTransactionInformationToFlatTrace(
...
-            .orElse("CREATE")
-            .toLowerCase(Locale.US);
-    builder.getActionBuilder().creationMethod(creationMethod);
+
+    // add creationMethod for create action
+    Optional.ofNullable(builder.getType())
+        .filter(type -> type.equals("create"))
+        .ifPresent(
+            __ -> {
+              final String creationMethod =
```
Comment (abdelhamidbakhta) [inline ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/results/tracing/flat/FlatTraceGenerator.java]: Why using a local variable there ? You could pass directly this value to the builder

File: ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/results/tracing/vm/VmTraceGenerator.java [hunk 2]
```diff
@@ -172,6 +180,10 @@ private void handleDepthIncreased(
         } else {
           if (currentTraceFrame.getPrecompiledGasCost().isPresent()) {
             op.setCost(op.getCost() + currentTraceFrame.getPrecompiledGasCost().get().toLong());
+          } else if (currentOperation.equals("STATICCALL")
+              && nextTraceFrame.map(TraceFrame::getDepth).orElse(0)
```
Comment (abdelhamidbakhta) [inline ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/results/tracing/vm/VmTraceGenerator.java]: Have you tested with several levels deep ?

Comment (matkt) [reply ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/results/tracing/vm/VmTraceGenerator.java]: I added a new test transaction with several levels deeps :  

- Block 0x17 with depth == 0
- Block 0x18 with depth == 2

File: ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/results/tracing/flat/FlatTraceGenerator.java (hunk 1 above)
Comment (matkt) [reply ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/results/tracing/flat/FlatTraceGenerator.java]: Done 👍

Comment (abdelhamidbakhta) [reply ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/results/tracing/flat/FlatTraceGenerator.java]: Nice

File: ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/results/tracing/vm/VmTraceGenerator.java (hunk 2 above)
Comment (abdelhamidbakhta) [reply ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/results/tracing/vm/VmTraceGenerator.java]: Nice

**Pull request merged**